Tools/WinMLRunner/BindingUtilities.h

```diff
@@ -144,6 +144,35 @@ namespace BindingUtilities
         return elementStrings;
     }
 
+    template <typename T>
+    void PreProcessImageToBinding(const SoftwareBitmap& softwareBitmap, ModelBinding<T>& binding, float scale, const std::array<float, 3>& meanStdDev)
+    {
+        const auto imgHeight = softwareBitmap.PixelHeight();
+        const auto imgWidth = softwareBitmap.PixelWidth();
+        const auto resultSize = imgHeight * imgWidth * 3;
+        if (binding.GetDataBufferSize() != resultSize)
+        {
+            throw hresult_invalid_argument(L"Image Input size/shape is different from what model expects");
```

Comment:
This will result in an unhandled exception here since this code path is not part of a try block.

IMHO it would be okay to throw and the calling code should try/catch these and report errors. Cases that cause unexpected program termination make sense to be handled with exceptions (and their exception messages should be reported to the user).

Additionally it would mean that other parts of the code like: 

        const TensorFeatureDescriptor& tensorDescription = description.try_as<TensorFeatureDescriptor>();

        if (!tensorDescription)
        {
            std::cout << "BindingUtilities: Input Description type isn't tensor." << std::endl;
            throw;
        }

Should directly throw and not write to the output directly (but instead rely on app level try/catch blocks to report errors to the user.